.pre-commit-config-cpp.yaml

```
@@ -0,0 +1,40 @@
+# This file contains pre-commit hooks for checking the C++ code. Since these
+# hooks require tools that may not trivial to install (unlike the Python hooks
+# like flake8), we keep these in a separate file to be applied manually.
+repos:
+-   repo: https://github.com/pocc/pre-commit-hooks
+    rev: python
+    hooks:
+      - id: clang-format
+        # Unclear if this is necessary, it seems like clang-format uses
+        # .clang-format if found by default, at least for newer versions. I'll
+        # specify this here just in case.
+        args:
+        - --style=file
+        - -i
+        exclude: '.*extern/.*'
```

Is the beginning `.*` necessary?
```suggestion
        exclude: 'extern/.*'
```